modin/numpy/arr.py

```diff
@@ -0,0 +1,333 @@
+        else:
+            if self.shape != other.shape:
+                # In this case, we either have two mismatched objects trying to do an operation
+                # or a nested 1D object that must be broadcasted trying to do an operation.
+                matched_dimension = None
+                if self.shape[0] == other.shape[0]:
+                    matched_dimension = 0
+                elif self.shape[1] == other.shape[1]:
+                    matched_dimension = 1
+                if matched_dimension is not None:
+                    if (
+                        self.shape[matched_dimension ^ 1] == 1
+                        or other.shape[matched_dimension ^ 1] == 1
+                    ):
+                        # caller = self if other.shape[matched_dimension ^ 1] == 1 else other
+                        # callee = other if other.shape[matched_dimension ^ 1] == 1 else self
+                        return (self, other, self._ndim, {"broadcast": True, "axis": 1})
+                else:
+                    raise ValueError(
+                        f"operands could not be broadcast together with shapes {self.shape} {other.shape}"
+                    )
+            else:
+                return (self, other, self._ndim, {"broadcast": False})
+
+    def _add(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        result = caller._query_compiler.add(callee._query_compiler, **kwargs)
+        return array(_query_compiler=result, _ndim=new_ndim)
+
+    __add__ = _add
+
+    def _divide(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        result = caller._query_compiler.truediv(callee._query_compiler, **kwargs)
+        if caller != self:
+            result = result.rtruediv(1)
+        return array(_query_compiler=result, _ndim=new_ndim)
+
+    __truediv__ = _divide
+
+    def _float_power(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        pass
+
+    def _floor_divide(
+        self,
+        x2,
+        out=None,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        if caller != self:
+            # No workaround possible until broadcasting fixed. GH#5529.
+            pass
+        result = caller._query_compiler.floordiv(callee._query_compiler, **kwargs)
+        if any(callee._query_compiler.eq(0).to_pandas()):
+            result = result.replace(numpy.inf, 0)
```

Comment:
could you pls explain _why_?

and why `any(foo.to_pandas())` instead of `foo.any()`?